### ISSUE TITLE
Expected value but found T_END at character 1.

### ISSUE THREAD

**Issue by Minilei89** [pdollar/coco#63]
Hi, so I'm currently trying to train deepmask using the COCO dataset but keep getting this T_END error. I'm wondering what are the main causes behind it and how to fix it. I'm currently running this on cuda 8.0 and cudnn 5.0 using Lua 5.2 and torch. If you need any more information, let me know.

Here is the stack:
[lhuang@localhost deepmask]$ th train.lua
Found Environment variable CUDNN_PATH = /usr/local/cuda/lib64/libcudnn.so.5-- ignore option rundir	
-- ignore option dm	
-- ignore option datadir	
-- ignore option gpu	
-- ignore option reload	
| running in directory /home/lhuang/deepmask/exps/deepmask/exp	
| number of paramaters trunk: 15198016	
| number of paramaters mask branch: 1608768	
| number of paramaters score branch: 526337	
| number of paramaters total: 17333121	
convert: data//annotations/instances_train2014.json --> .t7 [please be patient]
/home/lhuang/torch/install/bin/lua: ...e/lhuang/torch/install/share/lua/5.2/threads/threads.lua:183: [thread 2 callback] /home/lhuang/torch/install/share/lua/5.2/coco/CocoApi.lua:142: Expected value but found T_END at character 1
stack traceback:
	[C]: in function 'decode'
	/home/lhuang/torch/install/share/lua/5.2/coco/CocoApi.lua:142: in function '__convert'
	/home/lhuang/torch/install/share/lua/5.2/coco/CocoApi.lua:128: in function '__init'
	/home/lhuang/torch/install/share/lua/5.2/torch/init.lua:91: in function </home/lhuang/torch/install/share/lua/5.2/torch/init.lua:87>
	[C]: in function 'CocoApi'
	/home/lhuang/deepmask/DataSampler.lua:25: in function '__init'
	/home/lhuang/torch/install/share/lua/5.2/torch/init.lua:91: in function </home/lhuang/torch/install/share/lua/5.2/torch/init.lua:87>
	[C]: in function 'DataSampler'
	/home/lhuang/deepmask/DataLoader.lua:36: in function </home/lhuang/deepmask/DataLoader.lua:30>
	(...tail calls...)
	[C]: in function 'xpcall'
	...e/lhuang/torch/install/share/lua/5.2/threads/threads.lua:234: in function 'callback'
	/home/lhuang/torch/install/share/lua/5.2/threads/queue.lua:65: in function </home/lhuang/torch/install/share/lua/5.2/threads/queue.lua:41>
	[C]: in function 'pcall'
	/home/lhuang/torch/install/share/lua/5.2/threads/queue.lua:40: in function 'dojob'
	[string "  local Queue = require 'threads.queue'..."]:13: in main chunk
stack traceback:
	[C]: in function 'error'
	...e/lhuang/torch/install/share/lua/5.2/threads/threads.lua:183: in function 'dojob'
	...e/lhuang/torch/install/share/lua/5.2/threads/threads.lua:264: in function 'synchronize'
	...e/lhuang/torch/install/share/lua/5.2/threads/threads.lua:142: in function 'specific'
	...e/lhuang/torch/install/share/lua/5.2/threads/threads.lua:125: in function <...e/lhuang/torch/install/share/lua/5.2/threads/threads.lua:36>
	(...tail calls...)
	/home/lhuang/deepmask/DataLoader.lua:40: in function '__init'
	/home/lhuang/torch/install/share/lua/5.2/torch/init.lua:91: in function </home/lhuang/torch/install/share/lua/5.2/torch/init.lua:87>
	[C]: in function 'DataLoader'
	/home/lhuang/deepmask/DataLoader.lua:21: in function 'create'
	train.lua:101: in main chunk
	[C]: in function 'dofile'
	...uang/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: in ?
convert: data//annotations/instances_train2014.json --> .t7 [please be patient]

Any help would be appreciated, thanks!

**Comment by Jeff-sjtu** [comment]
I have the same problem while using CocoApi, Can you please to tell me how you fix this problem? Thanks!

**Comment by Minilei89** [comment]
The problem stems from the current torch package being unable to store data in CharStorage(). You will want to run an older version of torch. For me, I simply cloned the torch distro, used "git checkout 5961f52a65fe33efa675f71e5c19ad8de56e8dad", and installed everything normally from there. Make sure to do a clean install of torch. Check to make sure that your CharStorage() is working correctly by creating a text file and running 
> require 'torch'
> x = torch.CharStorage('hello.txt')
> = x

in lua. If it still comes out to 0, then it means CharStorage() is broken. If it works you should be told the value of X is non-zero.
